src/SwiftUI/SwiftUI/Color.cs

```diff
@@ -20,6 +20,25 @@ public unsafe partial class Color : View
 	{
 		internal IntPtr Data { get; private set; }
 
+		#region Static Colour Spaces
+		static IntPtr colorSpaceDisplayP3 = IntPtr.Zero;
+		internal static IntPtr ColorSpaceDisplayP3 {
+			get {
+				if (colorSpaceDisplayP3 == IntPtr.Zero) {
+					var fullMeta = RGBColorSpaceMetadata (0);
+					var meta = &fullMeta->Metadata.TypeDescriptor;
+					colorSpaceDisplayP3 = GetColorSpaceDisplayP3 (meta);
+				}
+				return colorSpaceDisplayP3;
+			}
+		}
+		static IntPtr colorSpacesRGB = IntPtr.Zero;
+		internal static IntPtr ColorSpacesRGB => GetColorSpacesRGB ();
+
+		static IntPtr colorSpacesRGBLinear = IntPtr.Zero;
+		internal static IntPtr ColorSpacesRGBLinear => GetColorSpacesRGBLinear ();
+		#endregion
+
 		#region Static Colours
 		static Color? black = null;
```

Comment:
I don't feel strongly about it, but I don't think we need to cache these

Reply:
I definitely feel the static Colors need caching because if used in a game or an app that needs to change colours frequently or similar scenarios, we don't want that object being re-created each time. Also these should be lazy loaded/created, so the overhead should just be 1st time they are used. You'll notice I didn't cache the `Color.Clear`, `Color.Primary` and `Color.Secondary` calls as these might change depending on theme or something else (can't think right now)